### PR TITLE
GCC support in CModule [1/N]

### PR DESCRIPTION
**Pull request by mephi42**
This is related to https://github.com/frida/frida-gum/pull/511 - Frida appears to be changed fairly often, and with my slow pace of adjusting the original PR, I keep running into merge conflicts. Would you consider taking GCC support piecemeal? In this PR I have preparatory commits that don't touch GumCModule.

### CODE REVIEW

File: bindings/gumjs/gumcmodule.c [hunk 1]
```diff
@@ -28,11 +28,20 @@ struct _GumCModule
   GumCModuleFinalizeFunc finalize;
 };
 
+enum _GumCModuleHeaderKind
+{
+  GUM_CMODULE_HEADER_TCC,
+  GUM_CMODULE_HEADER_FRIDA
+};
+
+typedef enum _GumCModuleHeaderKind GumCModuleHeaderKind;
```
Comment (oleavr) [inline bindings/gumjs/gumcmodule.c:37]: Let's move this typedef up to the other typedefs [*], and use `guint` as the type there.

[*] Their order should match the order of the struct/enum definitions below.

File: bindings/gumjs/meson.build [hunk 2]
```diff
@@ -150,11 +150,11 @@ gumjs_runtime_sources = [
...
-  'runtime/cmodule/inttypes.h',
-  'runtime/cmodule/stdint.h',
-  'runtime/cmodule/stdio.h',
-  'runtime/cmodule/stdlib.h',
-  'runtime/cmodule/string.h',
+  'runtime/cmodule-tcc/inttypes.h',
+  'runtime/cmodule-tcc/stdint.h',
+  'runtime/cmodule-tcc/stdio.h',
+  'runtime/cmodule-tcc/stdlib.h',
+  'runtime/cmodule-tcc/string.h',
```
Comment (oleavr) [inline bindings/gumjs/meson.build:157]: These will also have to be renamed in our Windows build system:
- bindings/gumjs/gumjs-common.props
- bindings/gumjs/gumjs-32.vcxproj.filters
- bindings/gumjs/gumjs-64.vcxproj.filters

(Careful search and replace should do, we can leave the testing to our CI.)

**Review by oleavr — CHANGES_REQUESTED** [review]
Thanks!  Sorry for the pain.  Things aren't usually as busy as they've been lately; I've been working on dependency upgrades and did some long-overdue refactoring to move Capstone, TinyCC, and QuickJS out to our SDK.

File: bindings/gumjs/gumcmodule.c (hunk 1 above)
Comment (mephi42) [reply bindings/gumjs/gumcmodule.c]: Ok.

File: bindings/gumjs/meson.build (hunk 2 above)
Comment (mephi42) [reply bindings/gumjs/meson.build:157]: Ok.

**Comment by oleavr** [comment]
Thanks a lot! (And apologies for the delay here, I didn't notice the update.)

**Pull request merged**